### PR TITLE
[ADD] 8.0 'sale_crm' migration;

### PR DESCRIPTION
**Pull request by legalsylvain**
Manage migration for : 
- sale_order.campaign_id from crm_lead.campaign_id (previously type_id);
- sale_order.medium_id from crm_lead.medium_id (previously channel_id)

See #211, for previous discussions.
CC : @StefanRijnhart, @christophlsa, @pedrobaeza .

Regards.

### CODE REVIEW

File: addons/sale_crm/migrations/8.0.1.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from openerp.openupgrade import openupgrade
+from openerp.modules.registry import RegistryManager
+
+logger = logging.getLogger('OpenUpgrade')
+
+
+def migrate_crm_lead_sale_order(cr, pool):
+    """Get the id of the crm that had generated a sale.order
+    If found:
+    - retrive crm_lead.type_id to set sale_order.campaign_id;
```
Comment (pedrobaeza) [inline addons/sale_crm/migrations/8.0.1.0/post-migration.py]: s/retrive/retrieve

File: addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---Fields in module 'sale_crm'---
+sale_crm     / account.invoice          / section_id (many2one)         : module is now 'sale' ('sale_crm')
+sale_crm     / res.users                / default_section_id (many2one) : module is now 'sales_team' ('sale_crm')
+# Nothing to do - fields moved and not changed
```
Comment (pedrobaeza) [inline addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt:4]: 'sale_crm' doesn't depend on 'sales_team', so it is not assured that this field is going to be kept before sales_team is marking it.

File: addons/sale_crm/migrations/8.0.1.0/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            # if found, get campaign_id and medium_id
+            execute(cr,
+                    'SELECT id, campaign_id, medium_id'
+                    ' FROM crm_lead'
+                    ' WHERE id=%s'
+                    ' AND partner_id=%s' % (crm_id, partner_id))
+            res = cr.fetchone()
+            if not res:
+                logger.warning(
+                    "Cannot retrive crm_lead.id '%s' "
```
Comment (pedrobaeza) [inline addons/sale_crm/migrations/8.0.1.0/post-migration.py]: retrieve

File: addons/sale_crm/migrations/8.0.1.0/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import re
+import logging
+
+from openerp.openupgrade import openupgrade
+from openerp.modules.registry import RegistryManager
+
+logger = logging.getLogger('OpenUpgrade')
+
+
+def migrate_crm_lead_sale_order(cr, pool):
```
Comment (pedrobaeza) [inline addons/sale_crm/migrations/8.0.1.0/post-migration.py]: You aren't using `pool` in the method

File: addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt (hunk 2 above)
Comment (legalsylvain) [inline addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt:4]: Thanks for your review,
Sales_team is a new module, installed on V8 if sale_crm is installed.
There is nothing to do.

**Comment by legalsylvain** [comment]
@pedrobaeza Is it all ok for you ?

Regards.

File: addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt (hunk 2 above)
Comment (pedrobaeza) [inline addons/sale_crm/migrations/8.0.1.0/openupgrade_analysis_work.txt:4]: I'm not confident with this, but if you have sucessfully migrated, I trust that the process is correct.

**Comment by pedrobaeza** [comment]
:+1:

File: addons/sale_crm/migrations/8.0.1.0/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import re
+import logging
+
+from openerp.openupgrade import openupgrade
+
+logger = logging.getLogger('OpenUpgrade')
+
+
+def migrate_crm_lead_sale_order(cr):
+    """Get the id of the crm that had generated a sale.order
```
Comment (StefanRijnhart) [inline addons/sale_crm/migrations/8.0.1.0/post-migration.py]: You can replace this whole method with a single query:

UPDATE sale_order sale
SET campaign_id = lead.campaign_id, medium_id = lead.medium_id
FROM crm_lead lead
WHERE sale.partner_id = lead.partner_id
    AND sale.origin like '% '||lead.id;

**Comment by legalsylvain** [comment]
Sorry, I used git --break-all function.
I close this PR, because I don't know how to fix it. 
New PR available here #265.
Sorry for the inconveniences.

**Pull request closed without merging**